File: client/python/osc_client.py
import socket
import json
import time
import argparse

class OSCClient:
    def __init__(self, host="127.0.0.1", port=12000):
        self.host = host
        self.port = port
        self.client = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
        print(f"OSC Client initialized: {host}:{port}")

    def send_osc(self, address, *args):
        """Send an OSC message"""
        message = json.dumps({"address": address, "args": args})
        self.client.sendto(message.encode(), (self.host, self.port))
        print(f"Sent OSC message: {address} {args}")

    def start_recording(self):
        """Start speech recording"""
        self.send_osc("/startRecording")

    def stop_recording(self):
        """Stop speech recording"""
        self.send_osc("/stopRecording")

    def close(self):
        """Close the socket connection"""
        self.client.close()

def main():
    parser = argparse.ArgumentParser(description='OSC Client for Speech to Text')
    parser.add_argument('--host', default='127.0.0.1', help='OSC server host')
    parser.add_argument('--port', type=int, default=12000, help='OSC server port')
    args = parser.parse_args()

    client = OSCClient(args.host, args.port)

    try:
        while True:
            print("\nCommands:")
            print("1. Start recording")
            print("2. Stop recording")
            print("3. Exit")
            
            choice = input("Enter your choice (1-3): ")
            
            if choice == "1":
                client.start_recording()
            elif choice == "2":
                client.stop_recording()
            elif choice == "3":
                break
            else:
                print("Invalid choice. Please try again.")
            
            time.sleep(0.1)  # Small delay to prevent CPU overuse

    except KeyboardInterrupt:
        print("\nExiting...")
    finally:
        client.close()

if __name__ == "__main__":
    main() 

File: client/javascript/osc_client.js
const WebSocket = require('ws');

class OSCClient {
    constructor(host = 'localhost', port = 8081) {
        this.host = host;
        this.port = port;
        this.ws = null;
        this.connected = false;
    }

    connect() {
        return new Promise((resolve, reject) => {
            this.ws = new WebSocket(`ws://${this.host}:${this.port}`);

            this.ws.on('open', () => {
                console.log('Connected to OSC bridge');
                this.connected = true;
                resolve();
            });

            this.ws.on('message', (data) => {
                try {
                    const message = JSON.parse(data);
                    console.log('Received:', message);
                } catch (error) {
                    console.error('Error parsing message:', error);
                }
            });

            this.ws.on('error', (error) => {
                console.error('WebSocket error:', error);
                reject(error);
            });

            this.ws.on('close', () => {
                console.log('Disconnected from OSC bridge');
                this.connected = false;
            });
        });
    }

    sendOSC(address, ...args) {
        if (!this.connected) {
            console.error('Not connected to OSC bridge');
            return;
        }

        const message = {
            address: address,
            args: args
        };

        this.ws.send(JSON.stringify(message));
        console.log('Sent OSC message:', message);
    }

    startRecording() {
        this.sendOSC('/startRecording');
    }

    stopRecording() {
        this.sendOSC('/stopRecording');
    }

    close() {
        if (this.ws) {
            this.ws.close();
        }
    }
}

// Example usage
async function main() {
    const client = new OSCClient();
    
    try {
        await client.connect();
        
        // Example: Start recording after 1 second
        setTimeout(() => {
            client.startRecording();
            
            // Stop recording after 5 seconds
            setTimeout(() => {
                client.stopRecording();
                
                // Close connection after 1 more second
                setTimeout(() => {
                    client.close();
                }, 1000);
            }, 5000);
        }, 1000);
        
    } catch (error) {
        console.error('Error:', error);
    }
}

// Run the example if this file is executed directly
if (require.main === module) {
    main();
}

module.exports = OSCClient; 